src/python/pants/option/global_options.py

```diff
@@ -1662,6 +1662,64 @@ class GlobalOptions(BootstrapOptions, Subsystem):
         ),
     )
 
+    _use_deprecated_pex_binary_run_semantics = BoolOption(
+        "--use-deprecated-pex-binary-run-semantics",
+        default=True,
+        help=softwrap(
+            """
+            If `true`, `run`ning a `pex_binary` will run your firstparty code by copying sources to
+            a sandbox (while still using a PEX for thirdparty dependencies). Additionally, you can
+            refer to the `pex_binary` using the value of its `entry_point` field (if it is a filename).
+
+            If `false`, `run`ning a `pex_binary` will build the PEX via `package` and run it directly.
+            This makes `run` equivalent to using `package` and running the artifact. Additionally,
+            the binary must be `run` using the `pex_binary`'s address, as passing a filename to `run`
+            will run the `python_source`.
+
+            Note that support has been added to Pants to allow you to `run` any `python_source`,
+            so setting this to `true` should be reserved for maintaining backwards-compatibility
+            with previous versions of Pants. Additionally, you can remove any `pex_binary` targets
+            that exist solely for running Python code (and aren't meant to be packaged).
+            """
+        ),
+        removal_version="2.15.0.dev0",
+        removal_hint=softwrap(
+            """
+            If `use_deprecated_pex_binary_run_semantics` is already set explicitly to `false`,
+            simply delete the option from `pants.toml` because `false` is now the default.
+
+            If set to `true`, removing the option will cause `run` on a `pex_binary` to package and
+            run the built PEX file. Additionally, the `pex_binary` must be referred to by its address.
+            To keep the old `run` semantics, use `run` on the relevant `python_source` target.
+            """
+        ),
+    )
+
+    @property
+    def use_deprecated_pex_binary_run_semantics(self) -> bool:
+        if self.options.is_default("use_deprecated_pex_binary_run_semantics"):
+            warn_or_error(
+                "2.14.0.dev1",
+                "the option --use-deprecated-pex-binary-run-semantics defaulting to true",
+                softwrap(
+                    f"""
+                    In Pants 2.14, by default, running a `pex_binary` will actually package the PEX
+                    and run it, as if you ran `package` followed by executing the built PEX.
```

Comment:
This confused me because I forgot what Pants version we're on. Maybe:

> Currently, running a `pex_binary` by default will not include the source files in the PEX, and will instead put them in a temporary sandbox. 
>
> In Pants 2.14, the default will change to instead build the PEX like you had run the `package` goal, and then execute that PEX. This is more consistent and intuitive behavior.